### PR TITLE
jujutsu: enable dynamic shell completions

### PR DESCRIPTION
**Pull request by junglerobba**
https://martinvonz.github.io/jj/latest/install-and-setup/#command-line-completion
In contrast to the current version, these will also autocomplete things like revisions, bookmarks etc
If there's a better way to generate them all at once by setting the environment variable to the respective shell, please let me know

- [x] Have you followed the [guidelines for contributing](https://github.com/Homebrew/homebrew-core/blob/HEAD/CONTRIBUTING.md)?
- [x] Have you ensured that your commits follow the [commit style guide](https://docs.brew.sh/Formula-Cookbook#commit)?
- [x] Have you checked that there aren't other open [pull requests](https://github.com/Homebrew/homebrew-core/pulls) for the same formula update/change?
- [x] Have you built your formula locally with `HOMEBREW_NO_INSTALL_FROM_API=1 brew install --build-from-source <formula>`, where `<formula>` is the name of the formula you're submitting?
- [x] Is your test running fine `brew test <formula>`, where `<formula>` is the name of the formula you're submitting?
- [x] Does your build pass `brew audit --strict <formula>` (after doing `HOMEBREW_NO_INSTALL_FROM_API=1 brew install --build-from-source <formula>`)? If this is a new formula, does it pass `brew audit --new <formula>`?

-----

### CODE REVIEW

File: Formula/j/jj.rb [hunk 1]
```diff
@@ -26,7 +26,13 @@ def install
 
     system "cargo", "install", *std_cargo_args(path: "cli")
 
-    generate_completions_from_executable(bin/"jj", "util", "completion", shell_parameter_format: :flag)
+    generate_completions_from_executable("env", "COMPLETE=bash", bin/"jj",
+      shells: [:bash], shell_parameter_format: :none)
+    generate_completions_from_executable("env", "COMPLETE=fish", bin/"jj",
+      shells: [:fish], shell_parameter_format: :none)
+    generate_completions_from_executable("env", "COMPLETE=zsh", bin/"jj",
+      shells: [:zsh], shell_parameter_format: :none)
```
Comment (carlocab) [inline Formula/j/jj.rb:34]: What checks for the `COMPLETE` variable? Is it `jj` itself, or some crate that they're using?

Comment (junglerobba) [reply Formula/j/jj.rb:34]: It’s a feature of the clap (clap_complete to be specific) crate: https://docs.rs/clap_complete/latest/clap_complete/env/index.html

**Comment by junglerobba** [comment]
Hey, any updates to this? Is this something that should be enabled already, or should it be up to users? The jj docs say that you shouldn't be using both types of completions simultaneously, so by including the non-dynamic completions as this package currently does it might make it harder for users to enable these dynamic ones themselves.

File: Formula/j/jj.rb (hunk 1 above)
Comment (daeho-ro) [reply Formula/j/jj.rb:34]: Maybe, we have to add `clap` style for the `shell_parameter_format` from `homebrew/brew` to handle it as general format.

Comment (daeho-ro) [reply Formula/j/jj.rb:34]: Now this should work.
```suggestion
    generate_completions_from_executable(bin/"jj", shell_parameter_format: :clap)
```

**Review by daeho-ro — APPROVED** [review]

**Comment by daeho-ro** [comment]
Thank you for your contribution, @junglerobba !

**Pull request merged**